tests/test_task.py

```diff
@@ -0,0 +1,184 @@
+from benchcab.task import Task
+from benchcab import internal
+from benchcab.benchtree import setup_fluxnet_directory_tree
+
+
+@pytest.fixture(autouse=True)
+def run_around_tests():
+    """`pytest` autouse fixture that runs around each test."""
+
+    # Setup:
+    if TMP_DIR.exists():
+        shutil.rmtree(TMP_DIR)
+    try:
+        TMP_DIR.mkdir()
+    except FileNotFoundError as err:
+        print(err, "\n")
+        print("Try running pytest from the project root directory.")
+        raise
+
+    # Run the test:
+    yield
+
+    # Teardown:
+    shutil.rmtree(TMP_DIR)
+
+
+def touch(path):
+    """Mimics the unix `touch` command."""
+    with open(path, 'a', encoding="utf-8"):
+        os.utime(path, None)
+
+
+def test_get_task_name():
+    """Tests for `get_task_name()`."""
+    # Success case: check task name convention
+    task = Task(1, "test-branch", "forcing-file.nc", "sci0", {"some_setting": True})
+    assert task.get_task_name() == "forcing-file_R1_S0"
+
+
+def test_get_log_filename():
+    """Tests for `get_log_filename()`."""
+    # Success case: check log file name convention
+    task = Task(1, "test-branch", "forcing-file.nc", "sci0", {"some_setting": True})
+    assert task.get_log_filename() == "forcing-file_R1_S0_log.txt"
+
+
+def test_get_output_filename():
+    """Tests for `get_output_filename()`."""
+    # Success case: check output file name convention
+    task = Task(1, "test-branch", "forcing-file.nc", "sci0", {"some_setting": True})
+    assert task.get_output_filename() == "forcing-file_R1_S0_out.nc"
+
+
+def test_fetch_files():
+    """Tests for `fetch_files()`."""
+
+    # Success case: fetch files required to run CABLE
+    task = Task(1, "test-branch", "forcing-file.nc", "sci0", {"some_setting": True})
+
+    # Setup mock namelists directory in TMP_DIR:
+    Path(TMP_DIR, internal.NAMELIST_DIR).mkdir()
+    touch(Path(TMP_DIR, internal.NAMELIST_DIR, internal.CABLE_NML))
+    touch(Path(TMP_DIR, internal.NAMELIST_DIR, internal.CABLE_SOIL_NML))
+    touch(Path(TMP_DIR, internal.NAMELIST_DIR, internal.CABLE_VEGETATION_NML))
+
+    setup_fluxnet_directory_tree([task], root_dir=TMP_DIR)
+
+    # Setup mock repository that has been checked out and built:
+    Path(TMP_DIR, internal.SRC_DIR, "test-branch", "offline").mkdir(parents=True)
+    touch(Path(TMP_DIR, internal.SRC_DIR, "test-branch", "offline", internal.CABLE_EXE))
+
+    task.fetch_files(root_dir=TMP_DIR)
+
+    assert Path(TMP_DIR, internal.SITE_TASKS_DIR,
+                task.get_task_name(), internal.CABLE_NML).exists()
+    assert Path(TMP_DIR, internal.SITE_TASKS_DIR,
+                task.get_task_name(), internal.CABLE_VEGETATION_NML).exists()
+    assert Path(TMP_DIR, internal.SITE_TASKS_DIR,
+                task.get_task_name(), internal.CABLE_SOIL_NML).exists()
+    assert Path(TMP_DIR, internal.SITE_TASKS_DIR,
+                task.get_task_name(), internal.CABLE_EXE).exists()
+
+
+def test_clean_task():
```

Comment:
Sounds good, I'll put any mock setup into a function.

Reply:
See commit: a98f5491be68872072050cba829aebe60d97ce65, I have made more than one function to do mock file manipulations so that it is roughly clear what each test does.